packages/teleport-project-generator-next/src/project-template.ts

```diff
@@ -26,5 +45,175 @@ export default {
       fileType: 'json',
     },
   ],
-  subFolders: [],
+  subFolders: [
+    {
+      name: 'components',
+      subFolders: [],
+      files: [
+        {
+          name: 'cms-item',
+          fileType: FileType.JS,
+          content: `import { useEffect, useState, useRef } from "react";
+/**
+* - @param loading - JSX to display when the componet is in loading state.
+* - @param error - JSX to display when the compnent hit a error state
+* - @param children - Use for the renderProps patter to call the parent
+* - @param fetcher - A promise from which we need to derive another three states, loading, error, data
+* - @param params - Params that need to be passed while fetching the resource.
+* - @param initialData - This is used to avoid the loading state, and fetch the new data when there is a change in the params
+*/
+
+const NO_INITIAL_DATA = Symbol(null);
+
+const CMSListItem = ({
+  loading = null,
+  error = null,
+  children,
+  fetcher,
+  params = null,
+  initialData = NO_INITIAL_DATA,
+}) => {
+  const missingInitialData = useRef(initialData === NO_INITIAL_DATA);
+  const [resource, setResourceStatus] = useState(
+    missingInitialData.current
+      ? {
+          state: "idle",
```

Comment:
Is this 'iddle' correct here? Should it be 'loading'?

Reply:
It moves to loading in this line
https://github.com/teleporthq/teleport-code-generators/pull/802/files/a0f52c42bc4b697162dd0fe36122b6231c0d6fda

The initial state is `idle`, because when the component receives `initialData` and `fetcher` both. For first render we don't make the API call.